edx_solutions_projects/receivers.py

```diff
@@ -31,3 +33,33 @@ def on_course_deleted(sender, **kwargs):  # pylint: disable=W0613
                 models.WorkgroupUser.objects.filter(workgroup=workgroup).delete()
                 models.Workgroup.objects.filter(id=workgroup.id).delete()
             models.Project.objects.filter(id=project.id).delete()
+
+
+@receiver(pre_delete, sender=WorkgroupUser)
```

Comment:
Considering the side-effect you mentioned, I guess it won't be a problem.

Otherwise, I think we can set the receiver to `django.contrib.auth.models.User` instead to catch the user being deleted from the system.

I'll confirm and get back to you.

Reply:
Thanks! I subscribed to this ticket, so I'll see the updates too :)